docs/concepts/what-is-a-retention-period.md

```diff
@@ -8,13 +8,19 @@ tags:
   - explanation
 ---
 
-A Retention Period is the amount of time a Workflow Execution Event History remains in the Cluster's persistence store.
+A Retention Period is the amount of time, in days, that a Workflow Execution Event History remains in the Cluster's persistence store.
 
-- [How to set the Retention Period for the Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace](/tctl/namespace/register/#--retention)
+- [How to set the Retention Period for a Namespace using an SDK](/application-development/features/#namespaces)
 
 A Retention Period applies to a single [Namespace](/concepts/what-is-a-namespace) and is set when the Namespace is registered.
 
-If the Retention Period isn't set, it defaults to 2 days.
+The Temporal Cluster triggers a Timer task at the end of the Retention Period that cleans up the associated Workflow Execution Event History on that Namespace.
```

Comment:
Not only event history, but also mutable state, current run record (if it is still current run), and visibility record. But those maybe too details, so I I would just say that it cleans up data for the associated Workflow Execution.